app/services/certificate.service.ts

```diff
@@ -0,0 +1,210 @@
+import { Injectable } from "@angular/core";
+import { List } from "immutable";
+import { AsyncSubject, Observable, Subject } from "rxjs";
+// tslint:disable-next-line:no-var-requires
+const forge = require("node-forge");
+
+import { Certificate } from "app/models";
+import { Constants, log } from "app/utils";
+import { BatchClientService } from "./batch-client.service";
+import {
+    BatchEntityGetter, BatchListGetter, ContinuationToken,
+    DataCache, EntityView, ListOptionsAttributes, ListView,
+} from "./core";
+import { ServiceBase } from "./service-base";
+
+export interface CertificateListParams {
+}
+
+export interface CertificateParams {
+    thumbprint?: string;
+    thumbprintAlgorithm?: string;
+}
+
+export interface CertificateListOptions extends ListOptionsAttributes {
+}
+
+export const defaultThumbprintAlgorithm = "sha1";
+
+export enum CertificateFormat {
+    pfx = "pfx",
+    cer = "cer",
+}
+
+@Injectable()
+export class CertificateService extends ServiceBase {
+    /**
+     * Triggered only when a certificate is added through this app.
+     * Used to notify the list of a new item
+     */
+    public onCertificateAdded = new Subject<string>();
+    public cache = new DataCache<Certificate>();
+
+    private _basicProperties: string = "thumbprint,thumbprintAlgorithm,state,stateTransitionTime";
+    private _getter: BatchEntityGetter<Certificate, CertificateParams>;
+    private _listGetter: BatchListGetter<Certificate, CertificateListParams>;
+
+    constructor(batchService: BatchClientService) {
+        super(batchService);
+
+        this._getter = new BatchEntityGetter(Certificate, this.batchService, {
+            cache: () => this.cache,
+            getFn: (client, params: CertificateParams) => {
+                const algorithm = params.thumbprintAlgorithm || defaultThumbprintAlgorithm;
+                return client.certificate.get(algorithm, params.thumbprint);
+            },
+        });
+
+        this._listGetter = new BatchListGetter(Certificate, this.batchService, {
+            cache: () => this.cache,
+            list: (client, params: CertificateListParams, options) => {
+                return client.certificate.list({ certificateListOptions: options });
+            },
+            listNext: (client, nextLink: string) => client.certificate.listNext(nextLink),
+        });
+    }
+
+    public get basicProperties(): string {
+        return this._basicProperties;
+    }
+
+    public list(options?: any, forceNew?: boolean);
```

Comment:
is there actually a listNext for certificates?

Reply:
Yes, there is nextLink in certificateListResult
![image](https://user-images.githubusercontent.com/26984399/37609231-d38ece24-2b59-11e8-8e84-846e7e80da6a.png)